CMakeLists.txt

```diff
@@ -44,6 +44,10 @@ else()
   # Use a Trilinos which has been found in an upstream project. This mechanism only supports the interface target.
   if(TARGET Trilinos::all_selected_libs)
     message(STATUS "Trilinos included from upstream project.")
+    add_library(_Trilinos_all_selected_libs INTERFACE)
+    target_link_libraries(
+      _Trilinos_all_selected_libs INTERFACE Trilinos::all_selected_libs
```

Comment:
I think this looks wrong. What you want is to pass on Trilinos when installing MIRCO, right? I think you need to do what you also did for OpenMP, i.e., add the respective find_package calls to a configurable cmake file.

Currently you are installing Trilinos again in the very last line, although it is already installed (it wouldn't have been found otherwise).

Reply:
Thanks for having a look at this!

Reply:
@RShaw026 can you try this? Are the next steps clear to you?

Reply:
@sebproell I am looking into it, but I have some doubts. The error we are getting now is when we are using `fetchcontent`. The configurable `cmake` file was created so that we could use the develop version of MIRCO. To use MIRCO via `fetchcontent` method, we do not even need to install the `mirco_lib` target. Is it possible to introduce a flag which tells MIRCO if it is not being built with other project? Then we can keep the install the install command inside the flag. 

Currently if I just remove `_Trilinos_all_selected_libs` from line ~43, the following error occurs:
CMake Error: install(EXPORT "mirco_libTargets" ...) includes target "mirco_lib" which requires target "_Trilinos_all_selected_libs" that is not in any export set.

Reply:
Have a look here: https://cmake.org/cmake/help/latest/guide/importing-exporting/index.html

We might need an `IMPORTED` target for Trilinos instead of it being an `INTERFACE` target.

Reply:
Okay, I am looking into it. Thanks.